cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -297,6 +298,83 @@ func createDNSService(dnsService *v1.Service, serviceBytes []byte, client client
 	return nil
 }
 
+func createCoreDNSConfigMap(client clientset.Interface, cm *v1.ConfigMap) error {
+	// Get the CoreDNS ConfigMap. If not present, Create a new default ConfigMap.
+	if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Get(kubeadmconstants.CoreDNSConfigMap, metav1.GetOptions{}); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil
+		}
+		if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Create(cm); err != nil {
+			if !apierrors.IsAlreadyExists(err) {
+				return errors.Wrap(err, "unable to create configmap")
+			}
+		}
+	}
+	// Get the Corefile and installed CoreDNS version.
+	corefile, currentInstalledCoreDNSVersion, err := GetCoreDNSInfo(client)
```

Comment:
`GetCoreDNSInfo()` can accept a `*ConfigMap` as an argument.
if `nil` it should fetch the CM, if non-`nil` it should use the given CM.
this avoids fetching the same CM twice.